x/coinswap/client/cli/tx.go

```diff
@@ -0,0 +1,278 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/cosmos/cosmos-sdk/x/coinswap/internal/types"
+)
+
+// Liquidity flags
+const (
+	MinReward = "min-reward"
+	MinNative = "min-native"
+	Deadline  = "deadline"
+	Recipient = "recipient"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	coinswapTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Coinswap transaction subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	coinswapTxCmd.AddCommand(client.PostCommands(
+		GetCmdAddLiquidity(cdc),
+		GetCmdRemoveLiquidity(cdc),
+		GetCmdBuyOrder(cdc),
+		GetCmdSellOrder(cdc))...)
+
+	return coinswapTxCmd
+}
+
+// GetCmdAddLiquidity implements the add liquidity command handler
+func GetCmdAddLiquidity(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-liquidity [deposit-coin] [deposit]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Add liquidity to the reserve pool",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Add liquidity to the reserve pool for a trading pair.
+			
+Example:
+$ %s tx coinswap add-liquidity dai 1000atom --min-reward 100atom --deadline 2h  --from mykey
```

Comment:
I think min-reward is pool-tokens not atoms

Reply:
Got it, will change this.